### DISCUSSION TITLE
Elsa Dashboard not work with Pascal case json

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#1526] [Q&A]
@sfmskywalker I have integrated Elsa in our existing project that was using Pascal Case for JSON responses. 
I got below error
![image](https://user-images.githubusercontent.com/52976818/133219285-083d250f-b5c6-45be-a449-2d73904731e4.png)

When I change JSON from Pascal to Camel it worked fine

Can you kindly let me know what change I need to do to support the pascal case for Elsa?

**Comment by sfmskywalker** [comment]
The JSON serializer settings that the Elsa Server API uses come from the [IEndpointContentSerializerSettingsProvider](https://github.com/elsa-workflows/elsa-core/blob/master/src/server/Elsa.Server.Api/Services/IEndpointContentSerializerSettingsProvider.cs) service.

There are two things you could do:

1. Replace the service implementation in the service collection with your own (there's a `Replace<TService, TImplementation>` extension method you could use for that.
2. Submit a PR that updates the [EndpointContentSerializerSettingsProvider](https://github.com/elsa-workflows/elsa-core/blob/master/src/server/Elsa.Server.Api/Services/EndpointContentSerializerSettingsProvider.cs) implementation that takes an options object that provides the serializer configuration.

**Reply by akghasemi** [reply]
Hi, I tried the first solution and implemented new class like but it is not working
 public class ElsaEndpointContentSerializerSettingsProvider : IEndpointContentSerializerSettingsProvider
    {
        private readonly JsonSerializerSettings _serializerSettings;

        public ElsaEndpointContentSerializerSettingsProvider()
        {
            _serializerSettings = new JsonSerializerSettings();
            _serializerSettings.ConfigureForNodaTime(DateTimeZoneProviders.Tzdb);
            _serializerSettings.NullValueHandling = NullValueHandling.Ignore;
            _serializerSettings.TypeNameAssemblyFormatHandling = TypeNameAssemblyFormatHandling.Simple;
            //_serializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver
            //{
            //    NamingStrategy = new CamelCaseNamingStrategy
            //    {
            //        ProcessDictionaryKeys = false,
            //        ProcessExtensionDataNames = true,
            //        OverrideSpecifiedNames = false
            //    }
            //};
            _serializerSettings.Converters.Add(new FlagEnumConverter(new DefaultNamingStrategy()));
        }

        public JsonSerializerSettings GetSettings() => _serializerSettings;
    }
and in startup :
  services.AddElsaApiEndpoints();
services.Replace<IEndpointContentSerializerSettingsProvider, ElsaEndpointContentSerializerSettingsProvider >(ServiceLifetime.Singleton);

but it simply ignores the new implementation and even doesn't call the new class constructor

**Reply by sfmskywalker** [reply]
I see now. There's a bit of an inconsistency with regards to what API controllers use which JSON settings.
Some controllers use the `IEndpointContentSerializerSettingsProvider`, while others use the JSON settings that are configured with ASP.NET itself.

To customize the latter, you can provide a lambda to the `AddElsaApiEndpoints` call. Example:

```csharp
services.AddElsaApiEndpoints(options => options.SetupNewtonsoftJson = json => json.SerializerSettings.ContractResolver = new DefaultContractResolver
{
    NamingStrategy = new DefaultNamingStrategy()
});
```
